### ISSUE TITLE
[reggen] Error check for data-intg-passthrough

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#9544]
The reggen supports an option called `data-intg-passthrough` on windowed regions.
If this variable is true, it basically means beyond the window region there is already data integrity, thus the data integrity should not be generated for it in reg_top (the integrity for registers will instead be moved to `tlul_adapter_reg`).

Having said that, the current scripting does not really handle / check the case if a reg block has multiple windows each with different expectations on `data-intg-passthrough`.  We do not have this case right now, but it would be good to explicitly add an error check to ensure all windows either agree or disagree.

**Comment by tjaychen** [comment]
@rswarbrick 
@msfschaffner 
@eunchan

**Comment by rswarbrick** [comment]
And/or teach the generated RTL to handle the mixed case properly. But I agree: that will probably never be used, so it's probably a better idea to mandate one or the other.

**Comment by msfschaffner** [comment]
Agreed. If we do not want to handle mixed cases in RTL (or don't have the time to implement that), let's just add a check for this.

**Comment by andreaskurth** [comment]
Triaged for `tlul`. Keeping https://github.com/lowRISC/opentitan/labels/Type%3AFutureRelease because we don't make use of this for M2.5 (as mentioned in the issue description).
